### PR TITLE
[CALCITE-3029] Java-oriented field type is wrongly forced to be NOT N…

### PR DESCRIPTION
**Pull request by zhztheplayer**
…ULL after being converted to SQL-oriented

JIRA: https://issues.apache.org/jira/browse/CALCITE-3029

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/jdbc/JavaTypeFactoryImpl.java [hunk 1]
```diff
@@ -239,27 +240,30 @@ public RelDataType toSql(RelDataType type) {
 
   /** Converts a type in Java format to a SQL-oriented type. */
   public static RelDataType toSql(final RelDataTypeFactory typeFactory,
-      RelDataType type) {
+                                  RelDataType type) {
```
Comment (zhztheplayer) [inline core/src/main/java/org/apache/calcite/jdbc/JavaTypeFactoryImpl.java]: I'll fix the indent.

File: core/src/test/java/org/apache/calcite/tools/PlannerTest.java [hunk 2]
```diff
@@ -703,6 +705,24 @@ private void runDuplicateSortCheck(String sql, String plan) throws Exception {
...
+   * SQL-oriented</a>. */
+  @Test public void testInsertSourceRelTypeWithNullValues() throws Exception {
+    Planner planner = getPlanner(null, Programs.standard());
+    SqlNode parse = planner.parse(
+        "insert into \"emps\" values(1, 1, null, 1, 1)");
+    SqlNode validate = planner.validate(parse);
+    RelNode convert = planner.rel(validate).rel;
+    RelDataType insertSourceType = convert.getInput(0).getRowType();
+    String typeString = SqlTests.getTypeString(insertSourceType);
+    assertEquals(typeString,
```
Comment (rubenada) [inline core/src/test/java/org/apache/calcite/tools/PlannerTest.java]: I think assertEquals parameters are inversed

Comment (zhztheplayer) [reply core/src/test/java/org/apache/calcite/tools/PlannerTest.java]: My mistake. Thanks!

File: core/src/main/java/org/apache/calcite/jdbc/JavaTypeFactoryImpl.java [hunk 3]
```diff
@@ -245,21 +244,24 @@ public static RelDataType toSql(final RelDataTypeFactory typeFactory,
...
-                field -> toSql(typeFactory, field.getType(), false)),
-              type.getFieldNames());
+      final RelDataType structured = typeFactory.createStructType(
+          type.getFieldList()
+              .stream()
+              .map(field -> toSql(typeFactory, field.getType(), false))
+              .collect(Collectors.toList()),
+          type.getFieldNames());
+      return mustSetNullability
+                 ? typeFactory.createTypeWithNullability(structured, type.isNullable())
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/jdbc/JavaTypeFactoryImpl.java]: Why we must have a flag `mustSetNullability` ?

Comment (rubenada) [reply core/src/main/java/org/apache/calcite/jdbc/JavaTypeFactoryImpl.java]: @danny0405 , possibly we will remove it. This is currently under discussion on the Jira ticket

Comment (zhztheplayer) [reply core/src/main/java/org/apache/calcite/jdbc/JavaTypeFactoryImpl.java]: Removed according to discussion in JIRA.

**Comment by rubenada** [comment]
LGTM

**Pull request merged**